### PR TITLE
drop EXPERIMENTAL_broadcast_tx_sync RPC method

### PR DESCRIPTION
**Pull request by telezhnaya**
This PR is a part of RPC tx methods redesign https://docs.google.com/document/d/1jGuXzBlMAGQENsUpy5x5Xd7huCLOd8k6tyMOEE41Rro/edit?usp=sharing

It has no usages on our RPC nodes, but I can't say anything about 3rd party nodes.
It is not covered by documentation in any of our sources.

We have 2 alternatives - `broadcast_tx_async`, `broadcast_tx_commit`. 
Both of them are covered with the doc https://docs.near.org/api/rpc/transactions
We have new method `send_tx` as well, I'll update the doc soon.

`EXPERIMENTAL_broadcast_tx_sync` adds ambiguity to the interface we provide. Moreover, it does not add any additional value - the execution guarantees are vague, the return type doesn't add anything comparing to existing methods. It's hard to image the use case where `EXPERIMENTAL_broadcast_tx_sync` would be much more convenient than existing methods.

### CODE REVIEW

**Review by jakmeier — APPROVED** [review]
Yes, makes sense to me. Let's get rid of it.

Maybe it should be included in a changelog?
`chain/jsonrpc/CHANGELOG.md` seems pretty dead but it still exists and might wants to be revived?
Otherwise, `CHANGELOG.md` under non-protocol changes would make sense to me, perhaps some people actually read through those.

**Comment by telezhnaya** [comment]
I suggest merging it after we introduce new `broadcast_tx` method. I'll update CHANGELOG mentioning the preferred option there

File: pytest/tests/sanity/rpc_tx_submission.py [hunk 1]
```diff
@@ -26,22 +26,21 @@
...
-        method_name = 'broadcast_tx_async' if i == 1 else 'EXPERIMENTAL_broadcast_tx_sync'
-        res = nodes[0].json_rpc(method_name,
-                                [base64.b64encode(tx).decode('utf8')])
-        assert 'error' not in res, res
-        time.sleep(5)
-        tx_hash = res['result'] if i == 1 else res['result']['transaction_hash']
-        tx_query_res = nodes[0].json_rpc('tx', [tx_hash, 'test0'])
-        assert 'error' not in tx_query_res, tx_query_res
-    time.sleep(1)
+tx = sign_payment_tx(nodes[0].signer_key, 'test1', 100 + i, i + 1, hash1)
```
Comment (nikurt) [inline pytest/tests/sanity/rpc_tx_submission.py:29]: `i` is undefined

(sounds philisophical :thinking: )

Comment (telezhnaya) [reply pytest/tests/sanity/rpc_tx_submission.py:29]: Wow, you've found a bigger problem - it looks like we never run this anywhere.
Do you know who's the best person to discuss adding this to our CI?

Comment (nikurt) [reply pytest/tests/sanity/rpc_tx_submission.py:29]: Nayduck runs it, and it failed in the most recent run.
https://nayduck.near.org/#/test/531124
https://github.com/near/nearcore/blob/9e12243b20afc2b1fecf1b8f14d9391c9469d039/nightly/pytest-sanity.txt#L118

Comment (telezhnaya) [reply pytest/tests/sanity/rpc_tx_submission.py]: Fixed issue with `i`

File: integration-tests/src/tests/nearcore/rpc_error_structs.rs [hunk 2]
```diff
@@ -387,7 +387,7 @@ fn test_tx_invalid_tx_error() {
                             let bytes = borsh::to_vec(&transaction_copy).unwrap();
                             spawn_interruptible(
                                 client
-                                    .EXPERIMENTAL_broadcast_tx_sync(to_base64(&bytes))
+                                    .broadcast_tx_commit(to_base64(&bytes))
```
Comment (VanBarbascu) [inline integration-tests/src/tests/nearcore/rpc_error_structs.rs:390]: This is call times out.

**Pull request merged**